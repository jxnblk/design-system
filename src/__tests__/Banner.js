import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { Banner, theme } from '..'

describe('Banner', () => {
  test('renders with no props other than theme', () => {
    const json = renderer.create(<Banner theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightGray)
    expect(json).toHaveStyleRule('color', theme.colors.text)
  })

  test('renders with default palette', () => {
    const json = renderer.create(<Banner palette='default' theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.palette.default.backgroundColor)
    expect(json).toHaveStyleRule('color', theme.palette.default.color)
  })

  test('renders with information palette', () => {
    const json = renderer.create(<Banner palette='information' theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.palette.information.backgroundColor)
    expect(json).toHaveStyleRule('color', theme.palette.information.color)
  })

  test('renders with success palette', () => {
    const json = renderer.create(<Banner palette='success' theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.palette.success.backgroundColor)
    expect(json).toHaveStyleRule('color', theme.palette.success.color)
  })

  test('renders with attention palette', () => {
    const json = renderer.create(<Banner palette='attention' theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.palette.attention.backgroundColor)
    expect(json).toHaveStyleRule('color', theme.palette.attention.color)
  })

  test('renders with warning palette', () => {
    const json = renderer.create(<Banner palette='warning' theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.palette.warning.backgroundColor)
    expect(json).toHaveStyleRule('color', theme.palette.warning.color)
  })

  test('renders with default palette alias', () => {
    const json = renderer.create(<Banner palette='gray' theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.palette.default.backgroundColor)
    expect(json).toHaveStyleRule('color', theme.palette.default.color)
  })

  test('renders with information palette alias', () => {
    const json = renderer.create(<Banner palette='blue' theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.palette.information.backgroundColor)
    expect(json).toHaveStyleRule('color', theme.palette.information.color)
  })

  test('renders with success palette alias', () => {
    const json = renderer.create(<Banner palette='green' theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.palette.success.backgroundColor)
    expect(json).toHaveStyleRule('color', theme.palette.success.color)
  })

  test('renders with attention palette alias', () => {
    const json = renderer.create(<Banner palette='orange' theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.palette.attention.backgroundColor)
    expect(json).toHaveStyleRule('color', theme.palette.attention.color)
  })

  test('renders with warning palette alias', () => {
    const json = renderer.create(<Banner palette='red' theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.palette.warning.backgroundColor)
    expect(json).toHaveStyleRule('color', theme.palette.warning.color)
  })
})