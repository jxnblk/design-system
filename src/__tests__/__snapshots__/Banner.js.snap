// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner only accepts preset colors 1`] = `
.c1 {
  text-align: flex-start;
}

.c2 {
  width: 100%;
  text-align: left;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  font-size: 14px;
}

.c3 {
  font-size: 16px;
  margin: 0px;
}

<div
  className=""
  color={undefined}
>
  <div
    className="c0 c1"
  >
    <div
      className="c2"
      width={1}
    >
      <h5
        className="c3"
        fontSize={2}
      />
      <span
        className="c4"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with blue bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #007aff;
  color: #fff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M11,7H9V5h2Zm0,8H9V9h2ZM10,0A10,10,0,1,0,20,10,10,10,0,0,0,10,0Z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with custom iconName and size 1`] = `
.c0 {
  background-color: #0a0;
  color: #fff;
}

.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12,1.54L14.55,9.6,23,9.53l-6.87,4.91,2.67,8-6.8-5-6.8,5,2.67-8L1,9.53,9.45,9.6Z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with green bg 1`] = `
.c0 {
  background-color: #0a0;
  color: #fff;
}

.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M8,15,3,10,4.41,8.59,8,12.17l7.59-7.59L17,6ZM10,0A10,10,0,1,0,20,10,10,10,0,0,0,10,0Z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with lightBlue bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #cdf;
  color: #049;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="darkBlue"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M11,7H9V5h2Zm0,8H9V9h2ZM10,0A10,10,0,1,0,20,10,10,10,0,0,0,10,0Z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with lightGreen bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #cec;
  color: #060;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="darkGreen"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M8,15,3,10,4.41,8.59,8,12.17l7.59-7.59L17,6ZM10,0A10,10,0,1,0,20,10,10,10,0,0,0,10,0Z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with lightOrange bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #fec;
  color: #950;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="darkOrange"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 22 19"
      width={24}
    >
      <path
        d="M12,12H10V8h2Zm0,4H10V14h2ZM0,19H22L11,0Z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with lightRed bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #fcc;
  color: #800;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="darkRed"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M10,0A10,10,0,1,0,20,10,10,10,0,0,0,10,0Zm1,11H9V5h2Zm0,4H9V13h2Z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with no props other than theme 1`] = `
.c0 {
  background-color: #0a0;
  color: #fff;
}

.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M8,15,3,10,4.41,8.59,8,12.17l7.59-7.59L17,6ZM10,0A10,10,0,1,0,20,10,10,10,0,0,0,10,0Z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with orange bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #f90;
  color: #fff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 22 19"
      width={24}
    >
      <path
        d="M12,12H10V8h2Zm0,4H10V14h2ZM0,19H22L11,0Z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with red bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #c00;
  color: #fff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M10,0A10,10,0,1,0,20,10,10,10,0,0,0,10,0Zm1,11H9V5h2Zm0,4H9V13h2Z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with text node 1`] = `
.c0 {
  background-color: #0a0;
  color: #fff;
}

.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12,1.54L14.55,9.6,23,9.53l-6.87,4.91,2.67,8-6.8-5-6.8,5,2.67-8L1,9.53,9.45,9.6Z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      >
        Header
      </h5>
      <span
        className="c6"
        fontSize={1}
      >
        <div
          className=""
        >
          Text
        </div>
      </span>
    </div>
  </div>
</div>
`;

exports[`Banner renders with text string 1`] = `
.c0 {
  background-color: #0a0;
  color: #fff;
}

.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12,1.54L14.55,9.6,23,9.53l-6.87,4.91,2.67,8-6.8-5-6.8,5,2.67-8L1,9.53,9.45,9.6Z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      >
        Header
      </h5>
      <span
        className="c6"
        fontSize={1}
      >
        Text
      </span>
    </div>
  </div>
</div>
`;
